### DISCUSSION TITLE
Lumi Run Beta

### DISCUSSION THREAD

**Discussion by JPSchellenberg** [Lumieducation/Lumi#1532] [Lumi Run]
Hey,

during the past weeks we were working on a new feature for Lumi:
Lumi Run allows you to host H5P on https://Lumi.run with a few clicks.

You can download the beta client here:
 [windows](https://lumirunbeta.s3.fr-par.scw.cloud/Lumi%20Setup%20Run-Beta-1.exe?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=SCWYJZVWN7QRCD7HV0RC%2F20210608%2Ffr-par%2Fs3%2Faws4_request&X-Amz-Date=20210608T165129Z&X-Amz-Expires=521965&X-Amz-Signature=1609b363f34c2adc9169ee1282234ac11b465f4680eda13cd78e69546f2f6c9f&X-Amz-SignedHeaders=host)
[macOS](https://github.com/Lumieducation/Lumi/releases/download/v0.8.0-beta.1/Lumi-0.8.0-beta.1.dmg)

For Feedback please send an email to c@Lumi.education or use the discussion here on GitHub.

![lumi-run](https://user-images.githubusercontent.com/9697985/117881461-8ffef500-b2a9-11eb-895f-f20f03b6d4db.gif)

![Screenshot 2021-05-11 at 22 26 53](https://user-images.githubusercontent.com/9697985/117881526-9ab98a00-b2a9-11eb-97a2-3296a45cef8b.png)

**Comment by damianduchamps** [comment]
Will you offer a data processing agreement for educational institutions for your hosting service as you process data of students on behalf of schools?

**Reply by sr258** [reply]
We are considering this, but the legal requirements of a data processing agreement require the service to be somewhat finalized, as changing the agreement isn't that easy after it has come into force. A data processing agreement that complies with Art. 28 GDPR also requires much more work on our side (i.e. we have to compile lists of our security measures as an addendum to the agreement) than a free service and means we make a long-term commitment. We'd also have to assist teachers / schools when students exercise their rights according to the GDPR. As a result, I don't think this could be offered as a free service anymore, but would mean we would have to charge for this "extra" feature. But still, it's something, I've though about and that might come in the future. At the moment, we simply consider ourselves the data controller and not the teachers.

If it helps: We don't collect the xAPI data about student usage (so we don't see what students entered) at the moment. The only thing that we see of students is their IP address when they connect to the server. Students must still send their usage data to you by downloading it and sending you the file. We do have ideas of adding a reporting tool to Lumi Run, but we haven't even started on this yet.

**Reply by otacke** [reply]
Please note that running a (commercial) service that offers H5P as a service may run into legal trouble as it would have to make sure it is not infringing on Joubel's H5P trademark. I know, here you're only storing files, but I am not a lawyer and don't know how to interpret those things from a legal perspective.

Please also keep in mind that H5P.com offering H5P as a service is the main source of income that funds Joubel's H5P development, so offering a competing service - even if not breaking trademark laws - may hurt H5P.

**Reply by sr258** [reply]
You're right about the fact that Joubel has a trademark on H5P, but Lumi and Lumi Run don't use H5P in its name for this reason. The technology itself is open source after all. A trademark doesn't rule out the usage of the term in descriptive texts.

**Reply by otacke** [reply]
As I said: not a lawyer.

**Reply by JPSchellenberg** [reply]
It is not our goal to compete with H5P or Joubel in any way. Our "mission" is to make learning accessible, individual and exciting. We noticed that it is too difficult for a lot of educators to host H5P so we try to offer a service that makes it easy for them and makes it more accessible. We are currently not planing on making it commercial. We would only charge money if the expenses on infrastructure become too expensive for us, which is very far away.

Besides that, I don't think that it would hurt H5P as even if we would make money we would also invest it into developing H5P. I mean look at git: git was developed by the linux team and GitHub used it to build a platform and a service that allows users to use git more efficiently. You wouldn't say that GitHub hurt the development of git.

**Reply by otacke** [reply]
That git analogy fails, because if H5P is supposed to git, then H5P.com would already be something like github (with non-free accounts only) - and gitlab would be competition for github. But I am not here to argue, and I am glad that you're reflecting on what you do.

**Reply by JPSchellenberg** [reply]
Arguing is not bad and I am thankful for the hint, but I think we see Lumi Run more as an alternative to a custom hosted wordpress than to H5P.com.

**Reply by otacke** [reply]
I see H5P.com as a hosting service - as an alternative to a custom hosted WordPress.

**Reply by JPSchellenberg** [reply]
Well, H5P.com is more directed to institutions as Lumi Run is more directed towards individuals.

H5P.com states:
> Our smallest package includes 1 - 3 authors and provides great value for small teams. If you are a teacher considering purchasing a license for your own personal use we encourage you to reach out to your employer to secure a license for yourself and some of your colleagues. The larger the license, the smaller the price per teacher.

So H5P.com does not really target individual educators, but the institutions or employers above them.
I don't know any teacher who got a license for a software from his employer if he is employed in a public school and that is an unnecessary barrier to the accessibility of H5P.

So there are two outcomes: 
1) Lumi Run fills the gap for individual educators.
2) H5P.com adapts and offers a solution for individuals.

Either way hosting H5P will become more accessible and the community will profit.

**Reply by otacke** [reply]
The perception of users and their intended use may be different to what you think. This doesn't sound like a request for individuals to me ("Will it be possible to host the service on a company or school server?", https://twitter.com/uivens/status/1392397851284619267).

**Reply by JPSchellenberg** [reply]
Ok, I see. I haven't really thought about offering something to schools or provide a hosting solution different to what Lumi Run is. That tweet was something I thought could be a good idea, but I don't see that we are going that way. Especially if it hurts H5P.com.

**Reply by sr258** [reply]
> The perception of users and their intended use may be different to what you think. This doesn't sound like a request for individuals to me ("Will it be possible to host the service on a company or school server?", https://twitter.com/uivens/status/1392397851284619267).

If a school or institution selfhosts Lumi Run, it's definitely not SaaS anymore and thus doesn't compete with h5p.com. In that sense it would be no different from the plugins for WordPress or Moodle.

**Reply by otacke** [reply]
It is a good idea! And I really like what you do - but I guess you know that already. Aaaand I don't _know_ if it's going to hurt H5P.com, especially given that it will become more attractive in the future with more exclusive features.

But I also know how appealing the price tag 0 € is, especially in the world of education - where people rather stick with a basic feature that's free instead of investing some money or effort to actually support projects. Guess you have not received too much exchange money as a small thank you for Lumi yourself ...

That's what I have been primed with during the last couple of years, hence my scepticism.

**Reply by JPSchellenberg** [reply]
It's not that no one wants to pay for those things, but that a lot of people can not pay for it. A lot of colleagues tried to buy software licenses (through the school) during the pandemic but all failed - not because there is no money (we will be buying a ton of student iPads), but the higher instances rejected their requests.

**Reply by otacke** [reply]
I am not reducing that problem to a monetary issue (discussed at https://www.olivertacke.de/2019/03/26/zahlt-ihr-getrennt-oder-zusammen/ already).

I deliberately mentioned investing _effort_. There are tons of things that one could do to support open source projects, see e.g. https://www.olivertacke.de/2017/05/19/open-source-software-voranbringen-auch-wenn-du-nicht-programmieren-kannst/ But, alas, even reporting a bug (to you by the way) seems to be too much effort ... https://h5p.org/comment/41577#comment-41577 But that's a completely different topic :-)

**Comment by DegrangeM** [comment]
Minor issue : You can paste the registration code with ctrl+V but you can't paste it with right click as the context menu doesn't show.

**Reply by DegrangeM** [reply]
After putting the code, I get a green message "Your were logged out." and nothing happen. It's possible I took too long to enter the code.

**Reply by JPSchellenberg** [reply]
Thank you for the feedback. The code is valid for 5 minutes, but I think the message should be clear about that and why the registration/login failed.

**Comment by os-tlo** [comment]
This is nice - is there a way to grab the embed code?

**Reply by JPSchellenberg** [reply]
Not right now, but I will add this as a feature-request.

**Reply by os-tlo** [reply]
Thank you!
